monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,145 @@
+
+    def build_test_payloads(self) ->List[DeliverablePayloads]: 
+        ''' This is the main method to process the test configuration and build RID payload object, maxium of one flight is allocated to each USS. '''
+        
+        usses = self.test_configuration.usses # Store the USS details 
+
+        all_test_payloads = [] # This holds the data that will be deilver
+        
+        test_reference_time = arrow.get(self.test_configuration.now)
+        test_start_time = arrow.get(self.test_configuration.test_start_time)
+        test_start_offset = test_start_time.shift(minutes =1)
+        test_start_offset_isoformat = test_start_offset.isoformat()
+
+        requested_flights = [] # This objects holds the modified aircraft state data, aircraft state data is read from disk and timestamps are modified to have recent ones. 
+        for state_data_index, current_disk_rid_state_data in enumerate(self.disk_rid_state_data): 
+            uss = usses[state_data_index]  
+
+
+            test_id = str(uuid.uuid4())
+            test_injection_path = '/tests/{test_id}'.format(test_id=test_id)
+            
+            disk_reference_time_raw = current_disk_rid_state_data['reference_time']
+            disk_reference_time = arrow.get(disk_reference_time_raw)
+
+            current_disk_rid_state_data['reference_time'] = test_reference_time.isoformat()
+            updated_timestamps_telemetry = []            
+
+            timestamp = test_start_offset.shift(seconds = 1)
+            
+            for telemetry_id, flight_telemetry in enumerate(current_disk_rid_state_data['flight_telemetry']['states']):
+
+                disk_telemetry_time_stamp_raw = flight_telemetry['timestamp']
+                disk_telemetry_time_stamp = arrow.get(disk_telemetry_time_stamp_raw)
+                reference_time_telemetry_time_stamp_delta_secs = (disk_telemetry_time_stamp - disk_reference_time).total_seconds()
+                
+                timestamp = timestamp.shift(seconds = reference_time_telemetry_time_stamp_delta_secs)
+                
+                flight_telemetry['timestamp'] = timestamp.isoformat()
+                updated_timestamps_telemetry.append(flight_telemetry)
+                        
+            flight_details =  current_disk_rid_state_data['flight_details']
+            operator_details = current_disk_rid_state_data['operator_details']
+            
+            operator_location = OperatorLocation(lat = operator_details['location']['lat'], lng = operator_details['location']['lng'])            
+            rid_flight_details = RIDFlightDetails(operator_id = str(uuid.uuid4()), operator_location = operator_location, operation_description = flight_details['operation_description'] , serial_number = flight_details['serial_number'], registration_number = flight_details['registration_number'])
+            test_flight_details = TestFlightDetails(effective_after = test_start_offset_isoformat,details = rid_flight_details)
```

Comment:
`effective_after` is declared as a `datetime`, but a `str` value is provided here

Reply:
I have modified the definition to have it as a string.